design/lakectl-testing.md

```diff
@@ -0,0 +1,102 @@
+# lakectl Testing
+
+## Purpose
+
+This document describes the design of a testing framework aimed at testing our CLI, namely `lakectl`.
+It specifies the requirements from the system and a set of technical questions we need to answer. Each question (in fact, each option), is accompanied by pros and cons, to help us understand which options will answer our needs the best. These are obviously not the only aspects to these decisions, and so, this document and the decisions derived from, are subject to reviews and suggestions.
+This document does not specify the cli tests to run, but the system intended to run them
+
+## Requirements
+* Usable both locally and as part of CI/CD
+* Ease of use - adding new tests should be as easy as possible. Moreover, adding new tests, or modifying existing ones, should not draw modifications to the testing framework itself
+  * It is obvious that changing the test **format** (e.g. expanding) will require framework modifications, in order to support the modified format
+* Decoupling of Framework and Data - the test data format should contain all the required info for the test itself, regardless of the framework design. In other words, we would like the ability to rewrite/replace the framework and still use the same tests data
+*
+
+## What We Want To Test 
```

Comment:
_Commented by @nopcoder_
how about output testing, so we like to verify specific format, colors and etc?

Reply:
Definitely.
I'm not sure how to achieve that, but it is definitely a requirement.
Adding to design doc